src/melee/vi/vi1202.c

```diff
@@ -20,7 +43,345 @@ void un_80321154(HSD_GObj* gobj)
     HSD_JObjAnimAll(GET_JOBJ(gobj));
 }
 
+void un_80321294(HSD_GObj* gobj)
+{
+    HSD_CObj* cobj = GET_COBJ(gobj);
+    HSD_CObjAnim(cobj);
+    if (cobj->aobj->curr_frame == 1.0F) {
+        vi_8031C9B4(0xe, 0);
+    }
+    if (cobj->aobj->curr_frame == cobj->aobj->end_frame) {
+        lb_800145F4();
+        gm_801A4B60();
+    }
+}
+
 void un_803218E0_OnFrame(void)
 {
     vi_8031CAAC();
 }
+
+extern char un_804A2F08[];
+extern vi1202_UnkStruct* un_804D7050;
+extern char mpLib_80458868[];
+
+void un_80321900(void)
+{
+    HSD_GObj* gobj = GObj_Create(0x16, 0x17, 0);
+    HSD_GObjProc_8038FD54(gobj, fn_803219AC, 0x13);
+    un_804D7050 = (vi1202_UnkStruct*)un_804A2F08;
+    un_80321950(un_804D7050);
+}
+
+void un_80321950(void* s)
+{
+    ((vi1202_UnkStruct*)s)->x0 = 0;
+    ((vi1202_UnkStruct*)s)->x4 = 0x10000;
+    ((vi1202_UnkStruct*)s)->x8 = 1.0F;
+    ((vi1202_UnkStruct*)s)->xC = 0;
+    ((vi1202_UnkStruct*)s)->x10 = *(s32*)((char*)Fighter_804D6500 + 0x20);
+    ((vi1202_UnkStruct*)s)->x14 = 0x83D60;
+    ((vi1202_UnkStruct*)s)->x18 = *(s32*)((char*)Fighter_804D6500 + 0x28);
+    ((vi1202_UnkStruct*)s)->x1C = 0;
+    ((vi1202_UnkStruct*)s)->x20 = 0;
+    ((vi1202_UnkStruct*)s)->x24 = 0;
+    ((vi1202_UnkStruct*)s)->x2C = -1;
+    ((vi1202_UnkStruct*)s)->x28 = -1;
+}
+
+void fn_803219AC(HSD_GObj* gobj)
+{
+    vi1202_UnkStruct* data = un_804D7050;
+    if (data->x4 < 0x10000) {
+        data->x4 = data->x4 + 1;
+    }
+    un_80321A00(gobj);
+    un_80321AF4(gobj);
+}
+
+void un_80321AF4(HSD_GObj* gobj)
+{
+    char* mpLib = mpLib_80458868;
+    HSD_GObj* cur;
+    vi1202_UnkStruct* data = un_804D7050;
+    s32 old_x24 = data->x24;
+    s32 flag = 0;
+    Vec3 pos;
+    char pad[12];
+
+    data->x24 = 0;
+    cur = *(HSD_GObj**)((char*)HSD_GObj_Entities + 0x20);
```

Comment:
Access the proper field or use `M2C_FIELD`. By the way, if you are using m2c locally, pass it `--valid-syntax` and it will generate `M2C_FIELD` usages when it can't determine the proper field.